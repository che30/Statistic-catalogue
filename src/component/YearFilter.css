.categories{
  margin-left: 57%;
  position: absolute;
  top: 20px;
}
@media(max-width: 768px){
  .categories{
    margin-left: 50%;
  }
}
@media(max-width: 567px){
  .categories{
    margin-left: 47%;
  }
}