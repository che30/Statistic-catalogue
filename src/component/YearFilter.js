/* eslint-disable jsx-a11y/label-has-associated-control */
import React from 'react';
import PropTypes from 'prop-types';
import './YearFilter.css';

const YearFilter = ({ changeFilter }) => {
  const categories = ['All', '2000-2002', '2003-2005',
    '2006-2008', '2009-2011', '2012-2014', '2016-2018', '2020-2021'];
  return (
    <div className="categories">
      <label>
        <select className="catFilter" onChange={(e) => { changeFilter(e.target.value); }}>
          {categories.map((cat) => (<option key={cat} value={cat}>{cat}</option>))}
        </select>
      </label>
    </div>

  );
};
YearFilter.propTypes = {
  changeFilter: PropTypes.func.isRequired,
};
export default YearFilter;
