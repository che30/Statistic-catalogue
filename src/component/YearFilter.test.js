import React from 'react';
import renderer from 'react-test-renderer';
import YearFilter from './YearFilter';

it('renders correctly', () => {
  const tree = renderer

    .create(<YearFilter />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});