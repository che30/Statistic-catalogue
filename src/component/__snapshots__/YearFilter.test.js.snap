// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="categories"
>
  <label>
    <select
      className="catFilter py-2 "
      onChange={[Function]}
    >
      <option
        value="All"
      >
        All
      </option>
      <option
        value="2000-2002"
      >
        2000-2002
      </option>
      <option
        value="2003-2005"
      >
        2003-2005
      </option>
      <option
        value="2006-2008"
      >
        2006-2008
      </option>
      <option
        value="2009-2011"
      >
        2009-2011
      </option>
      <option
        value="2012-2014"
      >
        2012-2014
      </option>
      <option
        value="2016-2018"
      >
        2016-2018
      </option>
      <option
        value="2020-2021"
      >
        2020-2021
      </option>
    </select>
  </label>
</div>
`;
